app/lib/features/home/providers/navigation.dart

```diff
@@ -0,0 +1,229 @@
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/features/home/data/models/nav_item.dart';
+import 'package:acter/features/home/widgets/custom_selected_icon.dart';
+import 'package:acter/main/routing/routing.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:acter/main/routing/routes.dart';
+
+const fallbackSidebarIdx = 1;
+const fallbackBottomBarIdx = 0;
+
+final spaceItemsProvider =
+    FutureProvider.family<List<SidebarNavigationItem>, BuildContext>(
+        (ref, context) async {
+  final spaces = ref.watch(spacesProvider);
+
+  return spaces.when(
+    loading: () => [
+      SidebarNavigationItem(
+        icon: const Icon(Atlas.arrows_dots_rotate_thin),
+        label: Text(
+          'Loading Spaces',
+          style: Theme.of(context).textTheme.labelSmall,
+          softWrap: false,
+        ),
+        location: null,
+      ),
+    ],
+    error: (error, stackTrace) => [
+      SidebarNavigationItem(
+        icon: const Icon(Atlas.warning_thin),
+        label: Text(
+          error.toString(),
+          style: Theme.of(context).textTheme.labelSmall,
+          softWrap: false,
+        ),
+        location: null,
+      )
+    ],
+    data: (spaces) {
+      spaces.sort((a, b) {
+        // FIXME probably not the way we want to sort
```

Comment:
The most traditional way to interact with data is to be done using ```StateNotifierProvider``` with State object (Space) as a param. But wit async crud, this usually won't be necessary.